semgrep/semgrep/output.py

```diff
@@ -305,6 +306,10 @@ def close(self) -> None:
 
         if self.final_error:
             raise self.final_error
+        elif self.rule_matches and self.settings.error_on_findings:
```

Comment:
The `--error` flag wasn't doing anything prior to this. It's functionality got lost somewhere along the line in a refactor.